src/CVRF/cvrf_eval.c

```diff
@@ -0,0 +1,423 @@
+/*
+ * Copyright 2017 Red Hat Inc., Durham, North Carolina.
+ * All Rights Reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <string.h>
+
+#include <libxml/xmlreader.h>
+#include <libxml/tree.h>
+#include <libxml/xpath.h>
+#include <libxml/parser.h>
+#include <libxml/xpathInternals.h>
+
+#include "public/cvrf.h"
+#include "cvrf_priv.h"
+
+#include "common/list.h"
+#include "common/_error.h"
+#include "common/xmltext_priv.h"
+#include "common/elements.h"
+#include "common/oscap_string.h"
+#include "common/util.h"
+
+#include "CPE/cpelang_priv.h"
+#include "CPE/public/cpe_dict.h"
+#include "CVSS/cvss_priv.h"
+#include "CVSS/public/cvss_score.h"
+
+#include "source/public/oscap_source.h"
+#include "source/oscap_source_priv.h"
+#include "source/public/oscap_source.h"
+
+#include "OVAL/public/oval_system_characteristics.h"
+#include "OVAL/public/oval_definitions.h"
+#include "OVAL/adt/oval_string_map_impl.h"
+#include "OVAL/public/oval_types.h"
+#include "OVAL/public/oval_probe_session.h"
+#include "OVAL/public/oval_probe.h"
+#include "OVAL/oval_definitions_impl.h"
+
+
+/*****************************************************************************
+ * Structure definitions
+ */
+
+struct cvrf_session {
+	struct cvrf_model *model;
+	char *os_name;
+	struct oscap_source *source;
+	char *export_file;
+	char *results_file;
+	struct oscap_stringlist *product_ids;
+	struct oval_definition_model *def_model;
+};
+OSCAP_ACCESSOR_STRING(cvrf_session, os_name);
+OSCAP_ACCESSOR_STRING(cvrf_session, export_file);
+OSCAP_ACCESSOR_STRING(cvrf_session, results_file);
+
+struct oscap_string_iterator *cvrf_session_get_product_ids(struct cvrf_session *session) {
+	return oscap_stringlist_get_strings(session->product_ids);
```

Comment:
Testing session before using it may seem too much now, but you never know how the code will evolve.